PowerPointLabs/PowerPointLabs/ShapesLab/CustomShapePane.cs

```diff
@@ -1677,6 +1712,41 @@ private void AddShapeButton_Click(object sender, EventArgs e)
 
             AddShapeFromSelection(selection, addIn);
         }
+
+        // A disabled button cannot respond to any events.
+        // Thus we register the event to the pane and when the mouse moves over
+        // the button, the tool tip will display.
+        private void CustomShapePane_MouseMove(object sender, MouseEventArgs e)
+        {
+            var parent = sender as Control;
+            if (parent == null)
+            {
+                return;
+            }
+            var ctrl = parent.GetChildAtPoint(e.Location);
+            if (ctrl != null)
+            {
+                if (ctrl.Visible && toolTip1.Tag == null)
+                {
+                    if (!_toolTipShown)
+                    {
+                        toolTip1.Show(ShapesLabText.AddShapeToolTip, ctrl, ctrl.Width / 2, ctrl.Height / 2);
+                        toolTip1.Tag = ctrl;
+                        _toolTipShown = true;
+                    }
+                }
+            }
+            else
+            {
+                ctrl = toolTip1.Tag as Control;
```

Comment:
Is it possible not to reuse this `ctrl` variable here? Issue is when I first read through this part of the code, I was wondering why you are checking `ctrl != null` again on line 1742 when you already did that in line 1727. This may cause unnecessary confusion to other developers in the future.